apps/crn-server/src/routes/research-outputs.route.ts

```diff
@@ -126,15 +123,22 @@ export const researchOutputRouteFactory = (
       const updateRequest = validateResearchOutputPutRequestParameters(body);
       validateResearchOutputPostRequestParametersIdentifiers(body);
 
+      const workingGroupOutput = updateRequest.workingGroups.length;
+
       const userRole = getUserRole(
         loggedInUser as UserResponse,
-        'teams',
-        updateRequest.teams,
+        workingGroupOutput ? 'workingGroups' : 'teams',
+        workingGroupOutput ? updateRequest.workingGroups : updateRequest.teams,
       );
 
-      // TODO: update the published value in hasEditResearchOutputPermission in
-      // display draft task. Currently we are not sending the value published
-      if (!loggedInUser || !hasEditResearchOutputPermission(userRole, false)) {
+      const result = await researchOutputController.fetchById(researchOutputId);
+      const publish = updateRequest.published && !result.published;
+
+      if (
+        !loggedInUser ||
+        !hasEditResearchOutputPermission(userRole, result.published) ||
+        (publish && !hasPublishResearchOutputPermission(userRole))
```

Comment:
I don't think you need this at all - the `hasEditResearchOutputPermission` seems to cover it